File: src/main/java/no/novari/flyt/integration/validation/constraints/ReferencedConfigurationIsCompleteValidator.java
package no.novari.flyt.integration.validation.constraints;

import no.novari.flyt.integration.validation.IntegrationValidationContext;
import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorContext;

public class ReferencedConfigurationIsCompleteValidator
        implements HibernateConstraintValidator<ReferencedConfigurationIsComplete, Long> {

    @Override
    public boolean isValid(Long value, HibernateConstraintValidatorContext hibernateConstraintValidatorContext) {
        IntegrationValidationContext integrationValidationContext = hibernateConstraintValidatorContext
                .getConstraintValidatorPayload(IntegrationValidationContext.class);

        return value == null || integrationValidationContext.getConfiguration().isCompleted();
    }

}

File: src/main/kotlin/no/novari/flyt/integration/validation/constraints/ReferencedConfigurationIsCompleteValidator.kt
package no.novari.flyt.integration.validation.constraints

import no.novari.flyt.integration.validation.IntegrationValidationContext
import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorContext

class ReferencedConfigurationIsCompleteValidator :
    HibernateConstraintValidator<ReferencedConfigurationIsComplete, Long> {
    override fun isValid(
        value: Long?,
        hibernateConstraintValidatorContext: HibernateConstraintValidatorContext,
    ): Boolean {
        val integrationValidationContext =
            hibernateConstraintValidatorContext.getConstraintValidatorPayload(
                IntegrationValidationContext::class.java,
            )

        return value == null || integrationValidationContext?.configuration?.completed == true
    }
}
